### ISSUE TITLE
Expose binDir of primary architecture in secondary architecture recipes

### ISSUE THREAD

**Issue by alaviss** [haikuports/haikuporter#170]
A lot of our recipe can't build with gcc2, but we also don't want the binaries to carry `secondaryArchSuffix` either (because secondary arch was only employed for gcc2, and the binary doesn't have different behavior per architecture).

A workaround that is widely employed in HaikuPorts is to define a variable like this:

```sh
commandBinDir=$binDir
if [ "$targetArchitecture" = "x86_gcc2" ]; then
	commandBinDir=$prefix/bin
fi
```

Then use this `commandBinDir` in place where you would use `binDir`. This works, but add some unnecessary noise, and could lead to massive `s|$prefix/bin|$newBinDir|g` if we wanted to change `binDir` location.

I propose that HaikuPorter expose this information directly through an environment variable. The name I'm thinking of right now is `primaryBinDir`, but I've no strong opinion for it.

**Comment by alaviss** [comment]
@fbrosson @korli It would be great if you could voice your opinions on this.

**Comment by fbrosson** [comment]
If the proposal is to have HaikuPorter automatically define `primaryBinDir=$prefix/bin` in **_all_** cases, i.e. without having to test if it is being called for x86_gcc2 and/or if the recipe is broken on x86_gcc2, then I'm OK with this proposal, with this name or with another name.
**_That said,_** I don't think this is **_that_** useful. I'd rather keep the code snippet we have right now, and wait for gcc2 to be retired.
Another reason for not implementing this is that:
1) If we choose to "convert" all the recipes which could benefit from it then this will add a lot of git noise.
2) if we choose to only use this new feature on new recipes or when we bump existing ones, then there will be 2 sets of recipes: those with the old style, and those with the new one. This will make it even harder to understand for newcomers.

I am much more in favor of using your recent idea, (without the tab):
```diff
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
 SECONDARY_ARCHITECTURES="x86"
+fi
```
to get rid of the `commandSuffix` everywhere where this is possible.

**Comment by korli** [comment]
Can you give an example of a recipe which would use the new variable?

**Comment by alaviss** [comment]
Here you go: https://github.com/haikuports/haikuports/search?q=commandBinDir&unscoped_q=commandBinDir

**Comment by fbrosson** [comment]
> Given that `gcc2` will only be deprecated in [R2](https://dev.haiku-os.org/milestone/R2), I would argue that it will be a long time before we can get rid of gcc2 artifacts in our tree.

Right, but I'm not yet convinced that having `primaryBinDir` (with all the git noise that could follow) would be an improvement.
That said, if @korli and/or other major contributors are OK with your proposal then I'm OK with it too.

**Comment by waddlesplash** [comment]
I agree with @fbrosson; the current solution is "OK for now", and the git noise will be more than the feature itself is worth.
